### PR TITLE
BC-2808 - BC-2975 - inline edit

### PR DESCRIPTION
**Pull request by christian-darsow**
# Short Description

- Implement inline editing for task card elements (title +text)
- Usage of the new Cards API [BC-2727-card](https://github.com/hpi-schul-cloud/schulcloud-server/tree/bc-2727-card)
- Usage of latest version of custom CK editor including autoformat plugin and highlighting configuration
- UI styling of dragging elements on TaskForm

## Links to Ticket and related Pull-Requests

BC-2808
https://github.com/hpi-schul-cloud/schulcloud-server/pull/3781
https://github.com/hpi-schul-cloud/dof_app_deploy/pull/436

## Checklist before merging

- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/store/task.unit.ts [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
 
 const mockTaskData = {
 	id: "123",
-	name: "abc",
-	description: { content: "Lorem ipsum", type: RichTextTypeEnum.RichtextCk5 },
+	cardElements: [],
+	draggable: true,
+	task: {
+		id: "456",
+		name: "",
```
Comment (MartinSchuhmacher) [inline src/store/task.unit.ts:10]: maybe we should fill in some more mockTaskData

File: src/store/task.ts [hunk 2]
```diff
@@ -1,33 +1,50 @@
...
+				id: "",
+				cardElementType: CardElementResponseCardElementTypeEnum.Title,
+				content: {
+					value: "",
+				},
+			},
+		],
+		draggable: true,
+		task: {
+			id: "",
```
Comment (MartinSchuhmacher) [inline src/store/task.ts:32]: more content would be nice as values

File: src/store/task.ts [hunk 3]
```diff
@@ -1,33 +1,50 @@
...
-	// TODO - implement/use proper type
-	taskData: TempTask = {
+	taskData: TaskCardResponse = {
 		id: "",
-		name: "",
-		description: { content: "", type: RichTextTypeEnum.RichtextCk5 },
+		cardElements: [
+			{
+				id: "",
+				cardElementType: CardElementResponseCardElementTypeEnum.Title,
```
Comment (MartinSchuhmacher) [inline src/store/task.ts:24]: what about a card element of type RichText?

File: src/pages/TaskForm.page.vue [hunk 4]
```diff
@@ -100,54 +92,105 @@ export default defineComponent({
...
+				cardElements.push({
+					id: title.value.id,
+					content: {
+						type: title.value.type,
+						value: title.value.model,
+					},
+				});
+				elements.value.forEach((element) => {
+					cardElements.push({
+						...(element.id && { id: element.id }),
```
Comment (MartinSchuhmacher) [inline src/pages/TaskForm.page.vue:179]: this nested logic seems bit much for one function, maybe outsource this into a separate function

File: src/pages/TaskForm.page.vue [hunk 5]
```diff
@@ -100,54 +92,105 @@ export default defineComponent({
 			},
 		];
 
-		const name = ref("");
-		const children = ref<Element[]>([]);
+		const title = ref<Element>({
+			id: "",
+			type: CardElementResponseCardElementTypeEnum.Title,
```
Comment (MartinSchuhmacher) [inline src/pages/TaskForm.page.vue:97]: `CardElementResponseCardElementTypeEnum` might be a bit too long to still make sense

File: src/components/task-form/TaskTitleElement.vue [hunk 6]
```diff
@@ -1,39 +1,77 @@
...
+::v-deep .v-textarea .v-input__slot {
+	padding: 0 !important;
+	margin-bottom: 0;
+}
+
+::v-deep .v-textarea textarea {
+	height: auto;
+	max-height: none;
+	padding: 0;
+	margin-top: var(--space-xl) !important;
```
Comment (MartinSchuhmacher) [inline src/components/task-form/TaskTitleElement.vue:69]: can we avoid using `!important` in new files somehow?

**Review by MartinSchuhmacher — COMMENTED** [review]
I am not sure about the naming of `TaskContentElement` and `TaskTextElement` in  regards to what these mean in comparison to the backend terms of `CardElements`. I would like to be consistent here so we do not use different names in BE and FE

File: src/components/task-form/TaskContentElement.vue [hunk 7]
```diff
@@ -1,56 +1,120 @@
 <template>
```
Comment (virgilchiriac) [inline src/components/task-form/TaskContentElement.vue:1]: rename file to CardElement

File: src/pages/TaskForm.page.vue [hunk 8]
```diff
@@ -5,27 +5,27 @@
 		headline="Task Form"
```
Comment (virgilchiriac) [inline src/pages/TaskForm.page.vue:5]: rename to TaskCard.page

File: src/components/task-form/TaskTextElement.vue [hunk 9]
```diff
@@ -0,0 +1,59 @@
+<template>
```
Comment (virgilchiriac) [inline src/components/task-form/TaskTextElement.vue:1]: rename file to RichTextCardElement

File: src/components/task-form/TaskTitleElement.vue [hunk 10]
```diff
@@ -1,39 +1,77 @@
 <template>
```
Comment (virgilchiriac) [inline src/components/task-form/TaskTitleElement.vue:1]: rename to TitleCardElement

File: src/store/task.ts [hunk 11]
```diff
@@ -1,33 +1,50 @@
 import { Action, Module, Mutation, VuexModule } from "vuex-module-decorators";
```
Comment (virgilchiriac) [inline src/store/task.ts:1]: rename file to task-card

File: src/store/types/task.ts [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import { TranslateResult } from "vue-i18n";
```
Comment (virgilchiriac) [inline src/store/types/task.ts:1]: rename file to CardElement

File: src/components/task-form/CKEditor.vue [hunk 13]
```diff
@@ -1,15 +1,13 @@
 <template>
```
Comment (virgilchiriac) [inline src/components/task-form/CKEditor.vue:1]: maybe the CKEditor components can be stored in another folder

**Review by virgilchiriac — CHANGES_REQUESTED** [review]

File: src/pages/TaskForm.page.vue (hunk 5 above)
Comment (christian-darsow) [reply src/pages/TaskForm.page.vue:97]: For now we keep this as provided by generated api

File: src/pages/TaskForm.page.vue (hunk 4 above)
Comment (christian-darsow) [reply src/pages/TaskForm.page.vue:179]: I agree, I will split that into several functions

File: src/store/task.unit.ts (hunk 1 above)
Comment (christian-darsow) [reply src/store/task.unit.ts:10]: yeap, good point

**Comment by virgilchiriac** [comment]
this PR is dimissed as it was simply moved to a diferent branch, so that the feature can be tested together and to match server and dof PRs
 see https://github.com/hpi-schul-cloud/nuxt-client/pull/2364

**Pull request closed without merging**